### PR TITLE
docs(docs): Add some "how-to" documentation for website development

### PR DESCRIPTION
**Pull request by Josmithr**
Includes notes on:
- "Documents" vs "Pages"
- Modifying sidebars
- Docs versioning

[AB#24262](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/24262)

### CODE REVIEW

**Comment by Josmithr** [comment]
For reviewers:

* I'm looking for feedback on what I have added so far. I've erred on the side of being somewhat minimal, and linking to Docusaurus's documentation, rather than re-stating what they have already documented. This also helps future-proof us somewhat as they update their own docs. That said, I'd still like for our own docs to be more useful than "go read Docusaurus's docs". It's a fine line to walk. Any feedback is appreciated s🙂
* I'm also looking for suggestions for other guides that would be helpful to add. We can always add more later as needed, but I'm also happy to add more now.

File: docs/README.md [hunk 1]
```diff
@@ -168,6 +168,65 @@ Rationale:
...
+
+1. Developer overhead - the more versions of the docs we offer, the more work we take on to keep documentation for supported versions up to date.
+2. User overhead - the more versions we offer, the more users have to think about when they engage with our website.
+3. Build performance - Docusaurus's build isn't fast. The more versions we add to our suite, the longer our build times become.
+
+#### Updating the Site Version
+
+These steps are based on Docusaurus's tutorial [here](https://docusaurus.io/docs/versioning#tutorials).
+
+Note: in order for API documentation to be available for the new version, you will need to ensure that API model artifacts are being published for the new major version series.
```
Comment (Josmithr) [inline docs/README.md:218]: @zhenmichael I'd like to document the steps for setting up API model artifacts publishing for a new release branch, if possible. Hoping you can help me fill that in 🙂

**Review by WayneFerrao — APPROVED** [review]
Reads well and I like the references to the Docusaurus docs!

**Review by alexvy86 — APPROVED** [review]
Pretty clear IMO 👍

File: docs/README.md (hunk 1 above)
Comment (zhenmichael) [reply docs/README.md:218]: Looks good overall! The pubishing pipeline should automatically generate new artifacts in the blob storage the first time a new major version release version is run. As long as the config file is updated the download-apis step should have access the correct artifacts.

**Review by zhenmichael — APPROVED** [review]

**Pull request merged**